src/ngraph_encapsulate_op.cc

```diff
@@ -350,15 +350,37 @@ class NGraphEncapsulateOp : public OpKernel {
                        << " Output Tensors freed: "
                        << output_tensors_bytes_free / (1024 * 1024) << " MB";
       }
-      m_ng_functions[signature] = ng_function;
+
+      try {
+        ng_exec = op_backend->compile(ng_function);
+      } catch (const std::exception& exp) {
+        BackendManager::UnlockBackend(m_op_backend_name);
```

Comment:
We are unlocking without locking. Not sure if we need to lock before compiling.

Reply:
Added locking before unlocking.